src/content/en/2021/caching.md

```diff
@@ -1,20 +1,488 @@
+
+The `immutable` directive is relatively new and can significantly improve cacheability for certain types of requests. However, it is still only supported by a few browsers, and we see most requests coming from from Host Networks like *Wix* with 16.4%, *Facebook* with 8.6%, *Tawk* with 2.8%, and *Shopify* with 2.4%.
+
+The most misused `Cache-Control` directive continues to be `set-cookie`, used for 0.07% of total *directives* for Desktop and 0.08% for Mobile. However, we are pleased to see a meaningful 0.16% reduction of usage from 2020.
+
+When we take a look when `no-cache`, `max-age=0` and `no-store` are used together, we also see a growing trend year after year, in which `no-store` is specified with either/both of `no-cache` and `max-age=0`, the `no-store` directive takes precedence, and the other directives are ignored. Driving more awareness around using these *directives* during larger conferences could help avoid accidentally wasted bytes.
+
+Fun fact: The most common `max-age` value is 30 days, and the largest value is 51 trillion years.
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="medium-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+### `304` Not Modified status
+
+When it comes to size, `304 Not Modified` responses are much smaller than `200 OK` responses, so it follows that page performance can be sped up by only delivering the necessary size of data. This is where correctly using conditional requests comes in because revalidation, and data savings can be done by using either an `Etag` or `Last-Modified` header.
+
+The `Last-Modified` response header works in conjunction with the `If-Modified-Since` request header to let the browser know if any changes have been made to the requested file.
+
+{{ figure_markup(
+  image="http-304-by-caching-strategy.png",
+  caption="HTTP 304 response rate by caching strategy.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1965033294&format=interactive",
+  sheets_gid="1136058277",
+  sql_file="valid_if_none_match_returns_304.sql"
+) }}
+
+We saw the distribution of 304 responses increase by 7.7% for `If-Modified-Since` between 2020 and 2021. This shows that the community is capitalizing on these data savings.
+
+### Validity of date strings
+
+The three main HTTP headers used to represent timestamps, `Date`,`Last-Modified` and `Expires` all use a date formatted string. The `Date` HTTP response header is almost always generated automatically by the web server, meaning that invalid values are extremely rare. Still, in the event that the date is set incorrectly it can affect cacheability on the response on which it is served.
+
+{{ figure_markup(
+  image="invalid-date-formats.png",
+  caption="Percent of responses with invalid date formats.",
+  description="TODO",
```

Comment:
description="A bar graph showing percentage of response headers with invalid date formats broken down by header. The percent of responses for desktop range from 2.8% for `Expires', 0.7% for `Last-Modified`, and 0.1% for `Date`. The percent of responses for mobile range from 3.2% for `Expires', 0.9% for `Last-Modified`, and 0.0% for `Date`.",